src/main/java/com/fasterxml/jackson/core/util/TextBuffer.java

```diff
@@ -790,6 +797,15 @@ public void append(String str, int offset, int len)
         } while (len > 0);
     }
 
+    private void validateAppend(int toAppend) {
```

Comment:
it may be more efficient to have a protected empty implementation in TextBuffer and to have this implementation as `protected final` in ReadConstrainedTextBuffer. SegmentedStringWriter is the only use case that I can think of for the plain TextBuffer though. Everything else uses ReadConstrainedTextBuffer.

Reply:
SegmentedStringWriter uses the append(char) method on TextBuffer and this method and its calculations will happen for every character.

Reply:
i think it should inline to the same code either way

Reply:
raw TextBuffer doesn't actually validate - only the ReadContrained one does. So this PR adds extra processing overhesd in the raw TextBuffer when it is not needed.

Reply:
no i think for raw TextBuffer it will simply inline validateAppend and the empty validateStringLength, detect that newTotalLength is unused, and eliminate the code inside validateAppend.

i'm 99% sure youd see this happen in a benchmark at least. however it's a bit less clear what will happen when there is a polymorphic call site.

Reply:
Right, the issue is that there are actually usages of polymorphic nature when `SegmentStringWriter` is used (unbounded). But I think bigger concern for me is actual usage of `append()` methods for validating case.

It does look like there are ideas of how to defer detection, however; and I agree that we do not require 100% byte/char accurate limits but rather order-of-magnitude matching.

Reply:
since the check is deferred now, i dont think it matters either way.